myosuite/envs/myo/walk_v0.py

```diff
@@ -134,4 +136,341 @@ def viewer_setup(self):
+        pelvis = self.sim.model.body_name2id('pelvis')
+        return np.array([self.sim.data.body_xpos[foot_id_l]-self.sim.data.body_xpos[pelvis], self.sim.data.body_xpos[foot_id_r]-self.sim.data.body_xpos[pelvis]])
+
+    def _get_forward_reward(self):
+        """
+        Gaussian that incentivizes a walking velocity. Going
+        over only achieves flat rewards.
+        """
+        vel = self._get_com_velocity()
+        if vel > 1.2:
+            return 1
+        else:
+            return np.exp(-np.square(1.2 - vel))
+
+    def _get_cyclic_rew(self):
+        """
+        Cyclic extension of hip angles is rewarded to incentivize a walking gait.
+        """
+        phase_var = (self._steps/self._hip_period) % 1
+        des_angles = np.array([0.8 * np.cos(phase_var * 2 * np.pi + np.pi), 0.8 * np.cos(phase_var * 2 * np.pi)], dtype=np.float32)
+        angles = self._get_angle(['hip_flexion_l', 'hip_flexion_r'])
+        return np.linalg.norm(des_angles - angles)
+
+    def _get_ref_deviation_cost(self):
+        """
+        Incentivize staying close to the initial reference orientation up to a certain threshold.
+        """
+        return np.exp(-max(0.0, np.linalg.norm(5.0 * (self.sim.data.qpos[3:7] - self._init_pos_root[3:7])) - 0.1))
+
+    def _get_torso_angle(self):
+        body_id = self.sim.model.body_name2id('torso')
+        return self.sim.data.body_xquat[body_id]
+
+    def _get_com_velocity(self):
+        """
+        Compute the center of mass velocity of the model.
+        """
+        mass = np.expand_dims(self.sim.model.body_mass, -1)
+        cvel = - self.sim.data.cvel
+        return (np.sum(mass * cvel, 0) / np.sum(mass))[4]
+
+    def _get_height(self):
+        """
+        Get center-of-mass height.
+        """
+        return self._get_com()[2]
+
+    def _get_rot_condition(self):
+        """
+        MuJoCo specifies the orientation as a quaternion representing the rotation
+        from the [1,0,0] vector to the orientation vector. To check if
+        a body is facing in the right direction, we can check if the
+        quaternion when applied to the vector [1,0,0] as a rotation
+        yields a vector with a strong x component.
+        """
+        # quaternion of root
+        quat = np.roll(self.sim.data.qpos[3:7], -1)
+        return [1 if np.abs(R.from_quat(quat).apply([1, 0, 0])[0]) > self._max_rot else 0][0]
+
+    def _get_com(self):
+        """
+        Compute the center of mass of the robot.
+        """
+        mass = np.expand_dims(self.sim.model.body_mass, -1)
+        com =  self.sim.data.xipos
+        return (np.sum(mass * com, 0) / np.sum(mass))
+
+    def _get_angle(self, names):
+        """
+        Get the angles of a list of named joints.
+        """
+        return np.array([self.sim.data.qpos[self.sim.model.jnt_qposadr[self.sim.model.joint_name2id(name)]] for name in names])
+
+    def _get_initial_state(self):
```

Comment:
This should go as a keyframe in the model. If you can specify the whole keyframe I can bake it into the model

Reply:
I changed the code such that it runs with keyframes now, you just have to integrate them into the model, such that I can get them with: model.key_qpos[1], model.key_qvel[1] and model.key_qpos[2], model.key.qvel[2]